backend/FwLite/MiniLcm/Validators/MultiStringValidator.cs

```diff
@@ -0,0 +1,12 @@
+﻿using FluentValidation;
+using MiniLcm.Models;
+
+namespace MiniLcm.Validators;
+
+internal static class MultiStringValidator
+{
+    public static IRuleBuilderOptions<T, MultiString> ValidMultiString<T>(this IRuleBuilder<T, MultiString> ruleBuilder)
+    {
+        return ruleBuilder.NotEmpty();
```

Comment:
this is probably incomplete, it really only blocks against having no data, we might want to tweak the name as some fields are allowed to be empty (eg CitationForm), it also does not block against an empty string as the WsId or an empty string as a value, we may or may not want that in the future. This is mostly here to demonstrate a single configuration of validation for a common type like MultiString

Reply:
Yeah, what a valid multi-string is, is an interesting problem:
- The WsId should be valid in the current project
- The value should be audio if it's an audio WS
- FLEx enforces a required lexeme form (for the primary vernacular WS, I think) when **creating** an entry, but you can just remove the lexeme form of a created entry.

So, I'm not so sure we'll want a generic method like this. Perhaps a "NotEmptyMultiString" for required values, like the name of a complex form type. But, guess what, you can just delete the name of a complex form type in FLEx (which causes it to default to "***"), so...we have to be careful not to be stricter than necessary.

Reply:
alright, I've made some changes. I'm going to talk about it is a json object. `Required` means there must be at least one field, and none of the values can be empty strings. Then there's `
NoEmptyValues` which just means none of the values can be empty strings, but it could be valid to have no keys at all, so that's what this is for. Example: LexemeForm should be required, but CitationForm would be `NoEmptyValues`

Audio and WsId can be handled later. WsId at the very least is back to a data dependent validation.